apluslms_roman/cli.py

```diff
@@ -365,20 +416,136 @@ def build_action(context):
+    try:
+        # save will return false if the file already exists and the document hasn't been changed
+        config = get_config(context)
+        if not config.save(overwrite=False):
+            exit(1, _("A project configuration already exists at {}".format(config.path)))
+    # config file was not found in folder
+    except FileNotFoundError:
+        try:
+            ProjectConfig.create(path_join(cwd, name))
+        except ValidationError as e:
+            exit(1, '\n'.join(render_error(e)))
+    print(_("Project configuration file {} created successfully.".format(name)))
+
+
 def config_print_action(context):
     all_ = not any(getattr(context.args, k, False) for k in ('user', 'project'))
     if context.args.debug:
         print("---\n# arguments:")
         data = {k: v for k, v in vars(context.args).items() if k[0] != '_' and not callable(v)}
         yaml_dump(data, stdout)
-    if all_ or context.args.user:
+    if all_ or context.args.global_:
         print("---\n# roman settings:")
         yaml_dump(context.settings._data, stdout)
     if all_ or context.args.project:
         print("---\n# project config")
         yaml_dump(get_config(context)._data, stdout)
 
 
+def config_set_action(context):
+    if context.args.global_ == context.args.project:
+        raise Exception("Choose either global settings or project settings")
+    document = context.settings if context.args.global_ else get_config(context)
+
+    for val in context.args.values:
+        try:
+            key, val = val.split('=', 1)
+            document.mlset_cast(key, val)
+        except ValueError as err:
+            if hasattr(err, 'value_type'):
+                raise ValueError(_("{} should be of type '{}', but was '{}'.").format(
+                    key, err.value_type, type(val).__name__))
+            else:
+                raise ValueError(_("Give values in format 'key=val'."))
+
+    document.validate()
+    document.save()
+
+
+def config_rm_action(context):
+    if context.args.global_ == context.args.project:
+        raise Exception("Choose either global settings or project settings")
+    document = context.settings if context.args.global_ else get_config(context)
+
+    for val in context.args.keys:
+        document.mldel(val)
+
+    document.validate()
+    document.save()
+
+
+def step_list_action(context):
+    steps = get_config(context).steps
+    num_len = max(2, len(str(len(steps)-1)))
+    name_len = max(4, max(len(s.get('name', '')) for s in steps))
+    header_fmt = "{:>%ds}  {:%ds} {}" % (num_len, name_len)
+    step_fmt = "{:%dd}. {:%ds} {}" % (num_len, name_len)
+    print(header_fmt.format('ID', 'NAME', 'IMAGE'))
+    for i, step in enumerate(steps):
+        print(step_fmt.format(i, step.get('name', ''), step['img']))
+    return
+
+
+def step_add_action(context):
+    args = context.args
+    env = args.env
+    if env:
+        try:
+            env = {a: b for a, b in [s.split('=', 1) for s in env]}
```

Comment:
Format `env = {a: b for a, b in (s.split('=', 1) for s in env)}` would be nicer as it uses generator instead of a list. Though, you could also do `env = dict(s.split('=', 1) for s in env)` or even better, `env = OrderedDict(s.split('=', 1) for s in env)`.